### ISSUE TITLE
Chrome DevTools window is always auto-opened for extension popup

### ISSUE THREAD

**Issue by imajus** [microsoft/vscode-js-debug#1958]
The problem occurs only when I'm running Chrome from VSCode for debugging. Whenever I click on the MetaMask Chrome extension icon, which triggers the MetaMask extension window to appear, the new DevTools window appears for the extension popup window.
Whilst this is handy for some projects, this is not needed and actually annoying for the project I'm currently working on but I couldn't find a way to disable that behavior.

![Screenshot from 2024-03-01 15-49-27](https://github.com/microsoft/vscode-js-debug/assets/1805633/591a80be-1e16-46c9-b634-725cf1bef383)

What I tried to do:
- Disabled the "Auto-open DevTools for popups" settings but it didn't help.
- Checked if I have any custom Chrome flags or Experiments activated, but found nothing which seems relevant.
- Tried to find a settings in VSCode which may control this behavior, but with no luck whatsoever.
- Tried to launch Chrome from a terminal, resembling the same command arguments I've found out the VSCode is using, and when running that way the problem is resolved:
  ```bash
  /opt/google/chrome/chrome --disable-background-networking --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-default-apps --disable-dev-shm-usage --disable-renderer-backgrounding --disable-sync --metrics-recording-only --no-first-run --no-default-browser-check --user-data-dir=/home/denis/.vscode/chrome about:blank
  ```

This leads me to an assumption that it's controlled over debugging protocol connection which is established from the VSCode after the browser is launched, and I don't know how to alter this behavior.

To summarize, my goal is to disable the automatic DevTools window opening for extension popups when running Chrome from VSCode for debugging.

**Comment by connor4312** [comment]
I'm guessing this is functionality in the metamask extension that automatically opens a devtools window when it's clicked if it detects the browser is being debugged. In the command line you're running: you're not enabling debugging via --remote-debugging-port/pipe, or attaching a debugger to it. This extension doesn't have any functionality to automatically open a devtools window (not even sure how I'd do that off the top of my head 😛)

**Comment by imajus** [comment]
I actually doubt that you're right. Because based on my knowledge there's no way to open a DevTools panel from the extension JavaScript execution environment, and there are [clear reasons](https://stackoverflow.com/questions/6801577/can-i-programmatically-open-the-devtools-from-a-google-chrome-extension) for such limitation.

As for the command I used to emulate Chrome execution for debugging I had to exclude the `--remote-debugging-pipe` argument because it didn't start that way. 
Now, thanks to your comment, I've attempted to include `--remote-debugging-pipe=9999` argument which allowed me to start Chrome in debugging mode:

```
/opt/google/chrome/chrome --disable-background-networking --disable-background-timer-throttling --disable-backgrounding-occluded-windows --disable-breakpad --disable-client-side-phishing-detection --disable-default-apps --disable-dev-shm-usage --disable-renderer-backgrounding --disable-sync --metrics-recording-only --no-first-run --no-default-browser-check --user-data-dir=/home/denis/.vscode/chrome --remote-debugging-port=9999 about:blank
``` 

But I still couldn't reproduce the issue that way, which may be considered as a proof of my point that it's actually VSCode controlling that behavior.

**Comment by imajus** [comment]
FYI I've made few other experiments to provide more info on the issue:

1. Reset `userDataDir` option in the VSCode debugging configuration, making the browser to start with a fresh settings.
2. Installed MetaMask from scratch
3. Installed another random Chrome extension with a popup functionality, namely Lighthouse

The results are the same as before, except that I've reproduced the issue with a different Chrome extension now.

![Screenshot from 2024-03-02 17-41-38](https://github.com/microsoft/vscode-js-debug/assets/1805633/da0b0908-060d-4eda-8456-8a61ab39d429)

![Screenshot from 2024-03-02 17-42-30](https://github.com/microsoft/vscode-js-debug/assets/1805633/1b76bff8-34ab-456d-9db7-00ad64fdc0e8)

**Comment by imajus** [comment]
FYI I've also just confirmed that the "Auto-open DevTools for popups" setting is toggled OFF in a new Chrome environment settings.

![Screenshot from 2024-03-02 17-47-54](https://github.com/microsoft/vscode-js-debug/assets/1805633/1d777a83-19dc-49ef-83b6-451f3bcdd1b4)

**Comment by connor4312** [comment]
You're right, this doesn't look specific to the metamask extension; I can also repro it in ublock origin and others who open those little popups. However, looking at traces I still don't think it's in js-debug: there's nothing we do that would request devtools get open. My guess is that there's functionality somewhere in Chromium that helpfully opens devtools if it detects that the browser is being run under automated software -- which would be useful for test runners for example.
